wdfn-server/waterdata/filters.py

```diff
@@ -138,3 +138,32 @@ def https_url(url):
     if not parsed.netloc:
         parsed = urlparse(f'//{url}')
     return ParseResult('https', parsed.netloc, parsed.path, *parsed[3:]).geturl()
+
+
+@app.template_filter('numerical_parameter_list')
+def numerical_parameter_list(parameter_group_series):
+    """
+    Generate text for a list of parameter codes.
+
+    :param list parameter_group_series: list of parameter series grouped by parameter group
+    :return: set of parameter codes
+    :rtype: str or None
+
+    """
+    series = chain.from_iterable([x['parameters'] for x in parameter_group_series])
+    # include only real-time parameters that have recent data
+    parameter_set = set(
+        [
+            s['parameter_code'] for s in series if 'Unit Values' in s['data_types'] and
+                                                                         (datetime.datetime.now().date() - datetime.timedelta(days=8) <=
+                                                                          s['end_date'].date() <= datetime.datetime.now().date() + datetime.timedelta(days=1))
+
+        ]
+    )
+
+    if parameter_set:
+        numerical_parameters = parameter_set
+    else:
+        numerical_parameters = None
+
+    return numerical_parameters
```

Comment:
I wonder if it is really necessary to return None for an empty set. Will returning an empty set work in the code?

Reply:
Good observation. Changed.